liblwgeom/lwgeom_geos.c

```diff
@@ -681,18 +718,26 @@ lwgeom_unaryunion(const LWGEOM *geom1)
 
 	initGEOS(lwnotice, lwgeom_geos_error);
 
-	g1 = LWGEOM2GEOS(geom1, 0);
+	g1 = LWGEOM2GEOS_PREC(geom1, LW_FALSE, op_precision);
 	if ( 0 == g1 )   /* exception thrown at construction */
 	{
 		lwerror("First argument geometry could not be converted to GEOS: %s", lwgeom_geos_errmsg);
 		return NULL ;
 	}
 
 	g3 = GEOSUnaryUnion(g1);
+	GEOSGeom_destroy(g1);
+
+	/* As of GEOS 3.6 r4118, GEOSUnaryUnion doesn't snap generated
```

Comment:
Was this filed upstream ? Worth having a link to the ticket, so to know when it is time to remove this check